### PR TITLE
Create power_of_two.cpp

### PR DESCRIPTION
**Pull request by neha-hasija17**
This Pull Request is for HacktoberFest 2020

#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1315"></a>

### CODE REVIEW

File: math/power_of_two.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        int result = n & (n-1);
+        if(result==0) 
+        
+        std::cout<<"Yes, the number "<<n<<" is a power of 2";
+        else 
+        std::cout<<"No, the number "<<n<<" is not a power of 2";
+             
+}
+
+/** Driver Code */
```
Comment (Panquesito7) [inline math/power_of_two.cpp:45]:
```suggestion
```

File: math/power_of_two.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    int n;
```
Comment (Panquesito7) [inline math/power_of_two.cpp:51]:
```suggestion
    int n = 0;
```

File: math/power_of_two.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
+
+/**
+ * @file power_of_two.cpp
```
Comment (Panquesito7) [inline math/power_of_two.cpp:3]:
```suggestion
 * @file
```

File: math/power_of_two.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
+
```
Comment (Panquesito7) [inline math/power_of_two.cpp:1]:
```suggestion
```

File: math/power_of_two.cpp [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+ * If N=23 then N-1 is 22, if we perform bitwise and of these two numbers then
+ * the result will not be zero , which indicates that it is not the power of 2
+ * \note This implementation is better than naive recursive or iterative approach. 
+ * 
+ * @author [Neha Hasija](https://github.com/neha-hasija17)
+ */
+ 
+#include <iostream>
+/**
+ * Function to test above algorithm
```
Comment (Panquesito7) [inline math/power_of_two.cpp:25]: Missing parameter documentation.
```suggestion
 * @brief Function to test above algorithm
 * @param n description
 * @returns void
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/neha-hasija17/C-Plus-Plus/actions

File: math/power_of_two.cpp [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ * If N= 32 then N-1 is 31, if we perform bitwise and of these two numbers then
+ * the result will be zero, which indicates that it is the power of 2
+ * If N=23 then N-1 is 22, if we perform bitwise and of these two numbers then
+ * the result will not be zero , which indicates that it is not the power of 2
+ * \note This implementation is better than naive recursive or iterative approach. 
+ * 
+ * @author [Neha Hasija](https://github.com/neha-hasija17)
+ */
+ 
+#include <iostream>
```
Comment (Panquesito7) [inline math/power_of_two.cpp:22]:
```suggestion
#include <iostream>  /// for std::cout
```

File: math/power_of_two.cpp [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+/**
```
Comment (Panquesito7) [inline math/power_of_two.cpp:23]:
```suggestion

/**
 * @namespace math
 * @brief Math algorithms
 */
namespace math {
/**
```

File: math/power_of_two.cpp [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        /// result stores the 
+        /// bitwise and of n and n-1
+        int result = n & (n-1);
+        if(result==0) 
+        
+        std::cout<<"Yes, the number "<<n<<" is a power of 2";
+        else 
+        std::cout<<"No, the number "<<n<<" is not a power of 2";
+             
+}
```
Comment (Panquesito7) [inline math/power_of_two.cpp:44]:
```suggestion
}
}  // namespace math
```

File: math/power_of_two.cpp [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    int n = 0;
+    ///n stores the input from the user
+    std::cout<<"enter a number "<<std::endl;
+    std::cin>>n;
+    ///function call with @param n
```
Comment (Panquesito7) [inline math/power_of_two.cpp:55]:
```suggestion
    ///  function call with @param n
```

File: math/power_of_two.cpp [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+             
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    int n = 0;
+    ///n stores the input from the user
```
Comment (Panquesito7) [inline math/power_of_two.cpp:52]:
```suggestion
    /// n stores the input from the user
```

File: math/power_of_two.cpp [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    std::cout<<"enter a number "<<std::endl;
+    std::cin>>n;
+    ///function call with @param n
+    power_of_two(n);
```
Comment (Panquesito7) [inline math/power_of_two.cpp:56]: Please add self-test cases, to ensure the algorithm works as excepted.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Comment by neha-hasija17** [comment]
@Panquesito7 please review this as well.

File: math/power_of_two.cpp [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+ * If N=23 then N-1 is 22, if we perform bitwise and of these two numbers then
+ * the result will not be zero , which indicates that it is not the power of 2
+ * \note This implementation is better than naive recursive or iterative
+ * approach.
+ *
+ * @author [Neha Hasija](https://github.com/neha-hasija17)
+ */
+
+#include <iostream>  /// for std::cout
+/**
```
Comment (Panquesito7) [inline math/power_of_two.cpp:25]:
```suggestion

/**
 * @namespace math
 * @brief Mathematical algorithms
 */
namespace math {
/**
```

File: math/power_of_two.cpp [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+     */
+    /// result stores the
+    /// bitwise and of n and n-1
+    int result = n & (n - 1);
+    if (result == 0) {
+        std::cout << "Yes, the number " << n << " is a power of 2";
+    } else {
+        std::cout << "No, the number " << n << " is not a power of 2";
+    }
+}
```
Comment (Panquesito7) [inline math/power_of_two.cpp:45]:
```suggestion
}
}  // namespace math

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: math/power_of_two.cpp [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    int n = 0;
+    /// n stores the input from the user
+    std::cout << "enter a number " << std::endl;
+    std::cin >> n;
+    /// function call with @param n
+    power_of_two(n);
```
Comment (Panquesito7) [inline math/power_of_two.cpp:65]:
```suggestion
    math::power_of_two(n);
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — APPROVED** [review]
LGTM, good work! 😄 👍

**Comment by neha-hasija17** [comment]
@Panquesito7 sir, it would be kind of you please merge it.

**Comment by Panquesito7** [comment]
> please merge it

Will do until @kvedala reviews the PR. 🙂

**Review by kvedala — CHANGES_REQUESTED** [review]
the contribution looks good.
However, please undo the changes to the file `pancake_sort.cpp`; then this can be merged

**Comment by Panquesito7** [comment]
> changes to the file `pancake_sort.cpp`

It seems those changes were made by the CI (https://github.com/TheAlgorithms/C-Plus-Plus/pull/1315/commits/caf2ecc54e0abc07f479d5eeb390f0614f9484eb) automatically.

**Comment by kvedala** [comment]
> > changes to the file `pancake_sort.cpp`
> 
> It seems those changes were made by the CI ([caf2ecc](https://github.com/TheAlgorithms/C-Plus-Plus/commit/caf2ecc54e0abc07f479d5eeb390f0614f9484eb)) automatically.

ah, ok

**Review by kvedala — APPROVED** [review]

**Pull request merged**